Conceptpower+Spring/src/main/java/edu/asu/conceptpower/web/ConceptEditController.java

```diff
@@ -180,11 +146,39 @@ public String confirmlEdit(@PathVariable("id") String id,
 	 * @return The list of existing concepts
 	 */
 	@RequestMapping(method = RequestMethod.GET, value = "conceptEditSynonymView")
-	public @ResponseBody
-	ConceptEntry[] searchConcept(@RequestParam("synonymname") String synonymname) {
-		ConceptEntry[] entries = conceptManager
-				.getConceptListEntriesForWord(synonymname.trim());
-		return entries;
+	public @ResponseBody ResponseEntity<String> searchConcept(@RequestParam("synonymname") String synonymname) {
+		ConceptEntry[] entries = conceptManager.getConceptListEntriesForWord(synonymname.trim());
+
+		StringBuffer jsonStringBuilder = new StringBuffer("{");

```

Comment:
Put the generation of Json at least in its own method. If you need it more often, maybe even a helper class.